internal/repo/github.go

```diff
@@ -0,0 +1,99 @@
+package repo
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/google/go-github/v55/github"
+	"github.com/zu1k/nali/internal/constant"
+	"io"
+	"net/http"
+	"regexp"
+)
+
+var (
+	client *github.Client
+)
+
+func getLatestRelease(owner, repo string) (*github.RepositoryRelease, error) {
+	client = github.NewClient(http.DefaultClient)
+	rel, resp, err := client.Repositories.GetLatestRelease(ctx, owner, repo)
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			// 404 means repository not found or release not found. It's not an error here.
+			return nil, fmt.Errorf("repository or release not found")
+		}
+		return nil, fmt.Errorf("API returned an error response: %s", err)
+	}
+	if rel == nil {
+		return nil, fmt.Errorf("repository release is nil")
+	}
+
+	return rel, nil
+}
+
+func getTargetAsset(rel *github.RepositoryRelease, assetFilters []*regexp.Regexp) *github.ReleaseAsset {
+	var tAsset *github.ReleaseAsset
+	found := false
+
+	for _, asset := range rel.Assets {
+		name := asset.GetName()
+
+		for _, filter := range assetFilters {
+			if filter.MatchString(name) {
+				tAsset = asset
+				found = true
+				break
+			}
+		}
+		if found {
+			break
+		}
+	}
+	return tAsset
+}
+
+func download(ctx context.Context, assetId int64) (data []byte, err error) {
+	var rc io.ReadCloser
+
+	rc, _, err = client.Repositories.DownloadReleaseAsset(ctx, constant.Owner, constant.Repo, assetId, http.DefaultClient)
```

Comment:
Do not use the default http client, which has no timeout control and does not use proxies.

See 
https://github.com/zu1k/nali/blob/f91569c50691f864bbca617d673315713e4602b1/pkg/common/httpclient.go#L30